src/components/Upload/UploadContext.tsx

```diff
@@ -0,0 +1,225 @@
+import React, {
```

Comment:
There's a lot going on in this file. Can it be broken up more? UploadManager state/context and the xhr request seem like easy picks to move to a different file.